pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -272,11 +268,28 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
 				})
 			})
 
-			Context("when the CAPI machine does exist", func() {
+			Context("when the CAPI machine and infra machine do exist", func() {
 				BeforeEach(func() {
 					By("Creating the CAPI machine")
 					capiMachine = capiMachineBuilder.Build()
 					Eventually(k8sClient.Create(ctx, capiMachine)).Should(Succeed(), "capi machine should be able to be created")
+
+					By("Creating the CAPI infra machine")
+					// we must set the capi machine as an owner of the capa machine
+					//  in order to ensure we reconcile capa changes in our sync controller.
+					capaMachine = capaMachineBuilder.WithOwnerReferences([]metav1.OwnerReference{
+						{
+							Kind:               machineKind,
+							APIVersion:         clusterv1.GroupVersion.String(),
+							Name:               capiMachine.Name,
+							UID:                capiMachine.UID,
+							BlockOwnerDeletion: ptr.To(true),
+							Controller:         ptr.To(false),
+						},
+					}).Build()
+
+					Expect(k8sClient.Create(ctx, capaMachine)).To(Succeed(), "capa machine should be able to be created")
```

Comment:
I am not sure about this move. I get here we need the MAPI owner refs to be set. But couldn't we keep the original `			By("Creating the CAPI infra machine")` in the outer Context and here only do an OwnerRef update for the capaMachine?

I worry this move would render the Context structure false